platform/ios/src/MGLMapView.h

```diff
@@ -329,6 +329,13 @@ MGL_EXPORT
  */
 @property (nonatomic, assign) CGPoint scaleBarMargins;
 
+/**
+ The maximum width ratio of the scale bar relative to the map view's frame.
+ This value is limited from 0.1 to 1.
+ Default is 0.5(half of map view's width).
+ */
+@property (nonatomic, assign) CGFloat scaleBarMaximumWidthRatio;
```

Comment:
What do you think if we rename this property to `scaleBarWidthRatio`?

Reply:
I think if changing this name, it might lead a wrong meaning. 
The value will only affect the maximum width of scale bar but could not equal to the actual width. 
The actual depends on the zoom level, and we can only say the scale bar's actual width is always less than or equal to the maximum width. 
@fabian-guerra